internal/hardware/validator.go

```diff
@@ -17,13 +18,38 @@ import (
 	"github.com/thoas/go-funk"
 )
 
+const (
+	tooSmallDiskTemplate   = "Disk is too small (disk only has %s, but %s are required)"
+	wrongDriveTypeTemplate = "Drive type is %s, it must be one of %s."
+)
+
+var (
+	diskEligibilityMatchers []*regexp.Regexp
+)
+
+func init() {
+	tmp := compile(tooSmallDiskTemplate, ".*", ".*")
+	diskEligibilityMatchers = append(diskEligibilityMatchers, tmp)
+
+	tmp = compile(wrongDriveTypeTemplate, ".*", ".*")
+	diskEligibilityMatchers = append(diskEligibilityMatchers, tmp)
+}
+
+func compile(template string, wildcards ...string) *regexp.Regexp {
```

Comment:
nit: this name is a bit too generic, maybe `compileDiskReasonTemplate` or something